onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -250,6 +250,18 @@ TensorrtLogger& GetTensorrtLogger() {
   return trt_logger;
 }
 
+std::unique_lock<OrtMutex> TensorrtExecutionProvider::GetEngineBuildLock() const {
+  static OrtMutex singleton;
+
+  // Acquire a lock only when force_sequential_engine_build_ is true;
+  return force_sequential_engine_build_ ? std::unique_lock<OrtMutex>(singleton) : std::unique_lock<OrtMutex>();
```

Comment:
I was wondering whether to change to use unique_lock all the time? Since build engine API is not thread safe.
(Even though current multithreading test doesn't fail here, it's much safer to put lock all the time)

Reply:
yes, i think we should always lock when creating engine. at the very least, we don't want multiple threads building at the same time as it can perturb the tactics timings.
btw, i noticed the buildEngineWithConfig() api is deprecated starting TRT 8.0
it's replaced by buildSerializedNetwork()
at some point, we need to make a pass through trt api calls and update any deprecated api calls. +@stevenlix fyi